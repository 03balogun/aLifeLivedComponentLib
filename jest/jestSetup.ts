/* eslint-disable no-undef */
jest.useFakeTimers('legacy'); // Stops reference error from appearing
