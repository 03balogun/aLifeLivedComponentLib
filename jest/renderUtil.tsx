import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';

function render(ui, { ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
