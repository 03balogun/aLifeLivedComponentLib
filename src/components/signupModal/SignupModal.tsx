import React from 'react';

// Components
import { Modal } from '../../themes/modal/Modal';
import { ModalContent } from '../../themes/modal/ModalContent';
import { ModalHeader } from '../../themes/modal/ModalHeader';
import { ModalFooter } from '../../themes/modal/ModalFooter';
import { Button } from '../button/Button';

type SignupModalProps = {
  onClose: () => void;
  isVisible: boolean;
  onSignupPress: () => void;
  onLoginPress: () => void;
};

export const SignupModal: React.FC<SignupModalProps> = ({
  onClose,
  isVisible,
  onSignupPress,
  onLoginPress,
}) => {
  return (
    <Modal onClose={() => onClose()} isVisible={isVisible}>
      <ModalHeader onHeaderPress={() => onClose()} />
      <ModalContent
        header="Welcome! ðŸ‘‹"
        description="To access all of A Life Lived please login or signup"
      />
      <ModalFooter>
        <Button
          title="Signup"
          onPress={() => onSignupPress()}
          width="50%"
          marginRight="s"
        />
        <Button
          title="Login"
          onPress={() => onLoginPress()}
          width="50%"
          marginLeft="s"
        />
      </ModalFooter>
    </Modal>
  );
};
