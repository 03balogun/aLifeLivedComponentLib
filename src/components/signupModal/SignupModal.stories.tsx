import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

// Component
import { SignupModal } from './SignupModal';

storiesOf('Modal', module)
  .addDecorator(Story => Story())
  .add('Signup', () => (
    <SignupModal
      isVisible={true}
      onClose={action('On close action fired')}
      onSignupPress={action('On signup press')}
      onLoginPress={action('On login press')}
    />
  ));
