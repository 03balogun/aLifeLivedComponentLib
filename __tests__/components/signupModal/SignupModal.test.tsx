import React from 'react';
import { render } from './../../../jest/renderUtil';

// Component
import { SignupModal } from '../../../src/components/signupModal/SignupModal';

const mockOnClose = jest.fn();
const mockOnSignupPress = jest.fn();
const mockOnLoginPress = jest.fn();

const setup = (isVisible = true) => {
  // Screen
  const utils = render(
    <SignupModal
      isVisible={isVisible}
      onClose={mockOnClose}
      onLoginPress={mockOnLoginPress}
      onSignupPress={mockOnSignupPress}
    />
  );

  return {
    ...utils,
  };
};

describe('component >> signupModal', () => {
  describe('Modal is visible', () => {
    test('Should render signup modal correctly', () => {
      const { toJSON } = setup();
      expect(toJSON()).toMatchSnapshot();
    });

    test('Should render correct text', () => {
      const { getByText } = setup();
      expect(getByText('Welcome!')).not.toBeNull();
      expect(
        getByText('To access all of A Life Lived please login or signup')
      ).not.toBeNull();
    });
  });

  describe('Modal is NOT visible', () => {
    test('not render modal', () => {
      const { toJSON } = setup(false);
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
