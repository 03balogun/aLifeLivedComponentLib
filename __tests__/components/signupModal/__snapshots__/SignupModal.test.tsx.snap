// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component >> signupModal Modal is NOT visible not render modal 1`] = `null`;

exports[`component >> signupModal Modal is visible Should render signup modal correctly 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "position": "absolute",
        "top": 0,
        "width": "100%",
        "zIndex": 100,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#252424",
          "borderRadius": 8,
          "display": "flex",
          "margin": 8,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "flex-start",
            "alignItems": "flex-start",
            "display": "flex",
            "margin": 8,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#f4f5f5",
                "display": "flex",
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 24,
              },
            ]
          }
        >
          X
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "display": "flex",
            "margin": 16,
            "paddingBottom": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#f4f5f5",
              "display": "flex",
              "fontSize": 24,
              "fontWeight": "bold",
              "lineHeight": 30,
              "textAlign": "center",
            },
          ]
        }
      >
        Welcome!
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#f4f5f5",
              "display": "flex",
              "fontSize": 16,
              "lineHeight": 20,
              "marginTop": 16,
              "textAlign": "center",
            },
          ]
        }
      >
        To access all of A Life Lived please login or signup
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "display": "flex",
            "flexDirection": "row",
            "margin": 16,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0FCFA6",
            "borderRadius": 8,
            "marginRight": 8,
            "opacity": 1,
            "padding": 16,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "display": "flex",
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Signup
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#0FCFA6",
            "borderRadius": 8,
            "marginLeft": 8,
            "opacity": 1,
            "padding": 16,
            "width": "50%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "display": "flex",
                "fontSize": 16,
                "fontWeight": "600",
                "lineHeight": 20,
                "textAlign": "center",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
